services/subscriber/http.go

```diff
@@ -1,39 +1,88 @@
 package subscriber
 
 import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 	"time"
 
-	"github.com/influxdata/influxdb/client/v2"
 	"github.com/influxdata/influxdb/coordinator"
 )
 
 // HTTP supports writing points over HTTP using the line protocol.
 type HTTP struct {
-	c client.Client
+	url         *url.URL
+	accessToken string
+	httpClient  *http.Client
 }
 
 // NewHTTP returns a new HTTP points writer with default options.
 func NewHTTP(addr string, timeout time.Duration) (*HTTP, error) {
-	conf := client.HTTPConfig{
-		Addr:    addr,
-		Timeout: timeout,
-	}
-	c, err := client.NewHTTPClient(conf)
+	u, err := url.Parse(addr)
 	if err != nil {
 		return nil, err
+	} else if u.Scheme != "http" && u.Scheme != "https" {
+		return nil, fmt.Errorf("Unsupported protocol scheme: %s, your address must start with http:// or https://", u.Scheme)
+	}
+	// Grab username from URL as token.
+	var token string
+	if u.User != nil {
+		token = u.User.Username()
+		u.User = nil
 	}
-	return &HTTP{c: c}, nil
+	return &HTTP{
+		url:         u,
+		accessToken: token,
+		httpClient: &http.Client{
+			Timeout: timeout,
+		},
+	}, nil
 }
 
 // WritePoints writes points over HTTP transport.
-func (h *HTTP) WritePoints(p *coordinator.WritePointsRequest) (err error) {
-	bp, _ := client.NewBatchPoints(client.BatchPointsConfig{
-		Database:        p.Database,
-		RetentionPolicy: p.RetentionPolicy,
-	})
-	for _, pt := range p.Points {
-		bp.AddPoint(client.NewPointFrom(pt))
+func (h *HTTP) WritePoints(p *coordinator.WritePointsRequest) error {
+	var b bytes.Buffer
+	for _, p := range p.Points {
+		if _, err := b.WriteString(p.String()); err != nil {
+			return err
+		}
+		if err := b.WriteByte('\n'); err != nil {
+			return err
+		}
+	}
+
+	u := *h.url
+	u.Path = "write"
+	params := url.Values{}
+	params.Set("db", p.Database)
+	params.Set("rp", p.RetentionPolicy)
+	params.Set("precision", "ns")
+	u.RawQuery = params.Encode()
+
+	req, err := http.NewRequest("POST", u.String(), &b)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "")
+	req.Header.Set("User-Agent", "InfluxDBSubscription")
+	if h.accessToken != "" {
+		req.Header.Set("InfluxDB-Access-Token", h.accessToken)
```

Comment:
After thinking on this a few days I want to change how this works to be more standard.

Instead of using the username + custom header to do the auth, I want to leave the InfluxDB side of things standard meaning. If you set the destination to `http://user:pass@example.com` then InfluxDB will set the standard Authorization header with BasicAuth encoding. 

Then the Kapacitor side will consume  this behavior by setting a special username like `_subscriber` and the password is the token.

This way using the `username:password` piece of a URL doesn't magically get set as a custom header causing confusion, for others using the subscription system.

@joelegasse Thoughts?